### PR TITLE
Add login aa1 to implement 1

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 

-   [x] Have you added an explanation of what your changes do and why you'd like them to be included?
-   [ ] Have you updated or added documentation for the change?
-   [x] Have you tested your changes with successful results?

**Type of Changes:**

 

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Documentation (non-breaking change which adds documentation)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

- Currently there is no login page.

**What is the new behavior (if this is a feature change)?**

- A login page has been implemented including error handling.

### CODE REVIEW

File: src/components/pages/LoginPage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  MDBCard,
+  MDBCardBody,
+  MDBInput,
+  MDBBtn,
+  MDBAlert,
+} from "mdbreact";
+//> Redux
+// Allows to React components read data from a Redux store, and dispatch actions
+// to the store to update data.
+import { connect } from "react-redux";
```
Comment (pinterid) [inline src/components/pages/LoginPage/index.jsx:24]:
```suggestion
import { connect } from "react-redux";

```

File: src/store/actions/authActions.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            code: 751,
+            message: "Auth failed for user.",
+            origin: "auth",
+          },
+        },
+      });
+    }
+  };
+};
+
```
Comment (pinterid) [inline src/store/actions/authActions.js:60]: Is there a newline required?

File: src/store/reducers/authReducer.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        action.payload.error.code,
+        action.payload.error.origin,
+        action.payload.error.message
+      );
+
+      return {
+        ...state,
+        authenticated: false,
+        error: action.payload.error,
+      };
```
Comment (pinterid) [inline src/store/reducers/authReducer.js:47]:
```suggestion
      };
      
```

**Review by pinterid — CHANGES_REQUESTED** [review]
Found some possible issues.

File: src/components/pages/LoginPage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  MDBCol,
+  MDBCard,
+  MDBCardBody,
+  MDBInput,
+  MDBBtn,
+  MDBAlert,
+} from "mdbreact";
+//> Redux
+// Allows to React components read data from a Redux store, and dispatch actions
+// to the store to update data.
```
Comment (kleberbaum) [inline src/components/pages/LoginPage/index.jsx]:
```suggestion
// Allows React components to read data , update data and dispatch actions
// from/to a Redux store.
```

File: src/components/pages/LoginPage/index.jsx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+  Pipelines,
+  User,
+  Project,
+  GitLabs,
+  Connectors,
+  Permissions,
+} from "../../organisms";
+//> Images
+import logoImg from "../../../assets/navigation/logo.png";
+// Too be added
```
Comment (kleberbaum) [inline src/components/pages/LoginPage/index.jsx]: Please remove this comment.

File: src/components/pages/LoginPage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+                    )}
+                    <MDBInput
+                      outline
+                      name="email"
+                      type="text"
+                      label="E-Mail"
+                      value={this.state.email}
+                      onChange={(e) =>
+                        this.setState({ [e.target.name]: e.target.value })
+                      }
```
Comment (kleberbaum) [inline src/components/pages/LoginPage/index.jsx]: I think username would be better.

File: src/store/actions/authActions.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+// Login user
+export const loginUser = (email, password) => {
+  return (dispatch, getState, { getIntel }) => {
+    if (email === "ci@s.co" && password === "cisco") {
```
Comment (kleberbaum) [inline src/store/actions/authActions.js]: I think username would be better.

File: src/store/actions/authActions.js (hunk 2 above)
Comment (kleberbaum) [reply src/store/actions/authActions.js:60]: hell yeah :)

File: src/store/reducers/authReducer.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
+// Have initial state for when state is not ready to be passed
```
Comment (kleberbaum) [inline src/store/reducers/authReducer.js]:
```suggestion
// Have initial state when state is not ready to be passed
```

**Review by kleberbaum — CHANGES_REQUESTED** [review]
Should be ok to merge after a few changes.

File: src/store/actions/authActions.js (hunk 2 above)
Comment (Aichnerc) [reply src/store/actions/authActions.js:60]: :)

**Review by schettn — APPROVED** [review]
lgtm

**Review by pinterid — APPROVED** [review]
LGTM

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

**Pull request merged**